backend/src/worker_flow.rs

```diff
@@ -159,24 +163,28 @@ pub async fn update_flow_status_after_job_completion(
             false
         };
 
-    let done = if !success || last_step || stop_early {
+    let done = if !(success || skip_loop_failures) || last_step || stop_early {
         let result = match new_status {
             FlowStatusModule::Success {
                 forloop_jobs: Some(jobs),
                 ..
             } => {
-                let mut results = Vec::new();
-                for job in jobs {
-                    let result = sqlx::query_scalar!(
-                        "SELECT result FROM completed_job WHERE id = $1 AND workspace_id = $2",
-                        job,
-                        w_id,
-                    )
-                    .fetch_optional(db)
-                    .await?
-                    .flatten();
-                    results.push(result.clone());
-                }
+                use futures::TryStreamExt;
+                let results = sqlx::query_as(
+                    "
+                  SELECT result
+                    FROM completed_job
+                   WHERE id = ANY($1)
+                     AND workspace_id = $2
+                     AND success = true
```

Comment:
does your autoformatter does this automatically ? this is nice

Reply:
No, I'm afraid I do this manually with my fingers :)